src/index.js

```diff
@@ -65,22 +90,22 @@ function createBonds(options) {
 		}
 	}
 
-	// TODO: Use more generic means to check on number, ideally push notification.
 	class SubscriptionBond extends Bond {
-		constructor(rpc) {
+		constructor(rpc, options = []) {
 			super();
-			this.rpc = rpc;
+			this.rpc = rpc();
+			this.options = [(_,n) => this.trigger(n), ...options];
 		}
 		initialise () {
-			api().subscribe(this.rpc, (_, n) => this.trigger(n))
-				.then(id => this.subscription = id);
+			// promise instead of id because if a dependency triggers finalise() before id's promise is resolved the unsubscribing would call with undefined
+			this.subscription = this.rpc.apply(api().pubsub, this.options);
 		}
 		finalise () {
-			api().unsubscribe(this.subscription);
+			this.subscription.then(id => api().pubsub.unsubscribe([id]));
 		}
 		map (f) {
-	        return new TransformBond(f, [this]);
-	    }
+					return new TransformBond(f, [this]);
```

Comment:
also, i had replaced this with the implementation from above. that should be used now:

```
map (f, outResolveDepth = 0, resolveDepth = 1) {
 	        return new TransformBond(f, [this], [], outResolveDepth, resolveDepth);
}
```